packit/source_git.py

```diff
@@ -486,23 +226,22 @@ def _populate_distro_dir(self):
             PackitException, if the dist-git repository is not pristine, that is,
             there are changed or untracked files in it.
         """
-        dist_git_repo = self.dist_git.local_project.git_repo
         dist_git_is_pristine = (
-            not dist_git_repo.git.diff() and not dist_git_repo.git.clean("-xdn")
+            not self.dist_git.git.diff() and not self.dist_git.git.clean("-xdn")
         )
         if not dist_git_is_pristine:
             raise PackitException(
                 "Cannot initialize a source-git repo. "
                 "The corresponding dist-git repository at "
-                f"{dist_git_repo.working_dir} is not pristine."
+                f"{self.dist_git.working_dir!r} is not pristine."
```

Comment:
Some more instructions on what to do would be welcome.
I `fedpkg clone`d a fresh dist-git repo, ran the `source-git init`, checked the source-git repo, `git reset`ed it and when I tried the `init` again I got the `not pristine` error.
So I checked the dist-git repo
- `git diff` - no change
- `git status` - ok, there's `BUILD/`, removed it

Ran `init` again, `not pristine` - but why when `git diff` and `git status` don't show anything?
`git clean -xdn` showed that there are still `BUILDROOT/`, `RPMS/`, `SRPMS/` and the downloaded tarball which need to be removed. But I wouldn't know if I hadn't checked the code.
(why those dirs don't show in `git status` even they are not .gitignored is a mystery to me)

Reply:
Valid concerns. Added a sentence about this.